### ISSUE TITLE
Python 3.x support

### ISSUE THREAD

**Issue by prock-fife** [fifengine/fifengine#631]
This issue was reported by **me@thomaskeller.biz**

It would be cool if FiFE would gain support for Python 3.1 and 3.2, since most distributions ship newer versions of Python these days with which FiFE is currently unable to cope with.

**Comment by Beliaar** [comment]
Please test the python3 branch, it was mostly automatically converted, and seems to work, even seems to still be compatible with python2.7, but i would not want to merge/release that without thorough testing.

**Comment by mbillingr** [comment]
Hi there, I started using FiFE with Python 3, which works smoothly but I had to manually adjust some implicit relative imports in `pychan/widgets/`: Imports of the form `from widget import Widget` no longer work in Python 3 and should be written as `from .widget import Widget`.

I don't think these files are automatically generated because they are part of the source tree, but then I don't know much about swig. So, did I make a mistake (maybe forgot to set a flag) when building the library?

Otherwise, if you agree that this should be changed I could create a pull request to the Python 3 branch. Changing these imports should be compatible with Python 2.7. Python 2.5 support would require additional `from __future__ import absolute_import` statements.

**Comment by Beliaar** [comment]
Please check the python3_futurize branch, that contains the python 2.7/python 3 compatible versions of the extension package now. And yes, those are not generated by swig.

**Comment by mbillingr** [comment]
That is exactly the branch that I used.

**Comment by Beliaar** [comment]
hmm... unless i messed up the last merge it should have the correct imports for python 3 - i accidentally pushed it, though, before i tested it.

**Comment by mbillingr** [comment]
After taking a closer look, some of the files  are fine and some still use the old imports.

For example, `slider.py` was changed 1 month ago to use explicit imports but in `spacer.py` the relevant imports have not been touched for 3 years.

**Comment by Beliaar** [comment]
It is possible that these are files that were added from another branch after i did the original conversion.

**Comment by mbillingr** [comment]
I see, thank you. Then I will use my manually patched files for now and look forward to your next commit :)

Sorry for hijacking the issue.

**Comment by Beliaar** [comment]
@fifengine/developers I'd like to ask that you try to write code that is already compatible with python 3 if possible - even on master. For a reference, check here - ignoring those things that need the future, builtins, past or six module when working in master: http://python-future.org/compatible_idioms.html

Once the futurize branch is merged into master you would have to do that one way or another, so you might as well get used to it now.

**Comment by LinuxDonald** [comment]
This is now "done". Python3 Support is in master. For problems with python3 we should open new tickets.
